app/furniture/marketplace/order_policy.rb

```diff
@@ -6,8 +6,13 @@ class Scope < ApplicationScope
       def resolve
         return scope.all if person.operator?
 
-        scope.joins(marketplace: [:room]).where(rooms: {space_id: person.spaces})
-          .or(scope.where(shopper: Shopper.where(person: person)))
+        if person.authenticated?
+          scope.where(rooms: {space_id: person.spaces})
+            .or(scope.where(shopper: Shopper.where(person: person)))
+        else
+          scope.where(shopper: Shopper.where(person: nil))
+            .where("marketplace_orders.created_at > ?", 1.week.ago)
+        end.joins(marketplace: [:room])
```

Comment:
I'm still a bit fuzzy on how authorization is wired up, but theoretically does this open access to any order by any Guest that might have any link to a recently created order?

Reply:
Yes, that is what happens. Order links are highly-unique; and a better solution would be to go through the Shopper.